### PR TITLE
(PC-5892) Stop using UserOfferer.rights before its removal

### PR DESCRIPTION
**Pull request by dbaty**
This attribute is not used anymore. On production, all instances of
UserOfferer are either 'editor' or 'admin'. The code only grants the
'editor' role. All in all, we are only interested in knowing whether a
User has an associated UserOfferer for a given Offerer.

Changes:

- `UserOfferer.rights` is not used anymore. The field is already
  nullable. We will remove it (through a migration) after the deployment
  of v123.

- The signature of `User.hasRights()` has been adapted.

- The signature of `Offerer.giveRights()` has been adapted.

- The signature of `ensure_current_user_has_rights()` has been adapted
  and the function has renamed as `check_user_has_rights_on_offerer()`
  for clarity (with the "check_" prefix for homogeneity).

Future: the field is already nullable and can removed once v123 is
deployed.

---

cf. pass-culture/pass-culture-pro#1082 pour l'arrêt de l'utilisation de champ côté portail pro.

### CODE REVIEW

File: src/pcapi/models/offerer.py [hunk 1]
```diff
@@ -42,13 +42,12 @@ class Offerer(
         String(9), nullable=True, unique=True
     )  # FIXME: should not be nullable, is until we have all SIRENs filled in the DB
 
-    def give_rights(self, user, rights):
+    def give_rights(self, user):
```
Comment (Akhilian) [inline src/pcapi/models/offerer.py:45]: On peux surement renommer ça au passage

```suggestion
    def grant_access(self, user):
```

File: src/pcapi/core/users/models.py [hunk 2]
```diff
@@ -181,22 +180,17 @@ def checkPassword(self, passwordToCheck):
     def get_id(self):
         return str(self.id)
 
-    def hasRights(self, rights, offerer_id):
+    def hasRights(self, offerer_id):
```
Comment (Akhilian) [inline src/pcapi/core/users/models.py:183]: Est-ce qu'on ne supprimerait pas complètement cette notion de droits ? :)
```suggestion
    def has_access(self, offerer_id):
```

File: tests/routes/pro/post_offerer_test.py [hunk 3]
```diff
@@ -124,7 +123,7 @@ def expect_the_current_user_to_be_editor_of_the_new_offerer(self, mock_api_entre
             # then
             assert response.status_code == 201
             offerer = Offerer.query.first()
```
Comment (Akhilian) [inline tests/routes/pro/post_offerer_test.py:125]: Vu que l'assertion change, je renommerais bien le nom du test qui parle toujours d'editor

**Review by Akhilian — APPROVED** [review]

File: src/pcapi/models/offerer.py (hunk 1 above)
Comment (dbaty) [reply src/pcapi/models/offerer.py:45]: J'aime beaucoup l'idée de renommer en `grant_access()` et `has_access()`. Bien vu, je prends !

**Pull request merged**